corehq/apps/app_manager/views/apps.py

```diff
@@ -714,6 +742,13 @@ def parse_sync_interval(interval):
     if should_edit("build_spec"):
         resp['update']['commcare-version'] = app.commcare_minor_release
 
+    if should_edit("practice_mobile_worker_id"):
+        user_id = hq_settings['practice_mobile_worker_id']
+        if user_id:
+            get_and_assert_practice_user_in_domain(user_id, request.domain)
+        if not has_privilege(request, privileges.PRACTICE_MOBILE_WORKERS):
+            app.practice_mobile_worker_id = None
```

Comment:
could swap these two if statements to avoid calling `get_and_assert_practice_user_in_domain` if the domain doesn't have privileges.